.github/ci-heapster.sh

```diff
@@ -0,0 +1,152 @@
+        echo "Command failed. Attempt $n/3:"
+      else
+        echo "The command has failed after $n attempts."
+        return 1
+      fi
+    fi
+  done
+}
+
+setup_dist_bins() {
+  is_exe "dist/bin" "saw" && is_exe "dist/bin" "jss" && return
+  extract_exe "saw" "dist/bin"
+  extract_exe "jss" "dist/bin"
+  export PATH=$PWD/dist/bin:$PATH
+  echo "$PWD/dist/bin" >> $GITHUB_PATH
+  strip dist/bin/saw* || echo "Strip failed: Ignoring harmless error"
+  strip dist/bin/jss* || echo "Strip failed: Ignoring harmless error"
+}
+
+install_z3() {
+  is_exe "$BIN" "z3" && return
+
+  case "$RUNNER_OS" in
+    Linux) file="ubuntu-16.04.zip" ;;
+    macOS) file="osx-10.14.6.zip" ;;
+    Windows) file="win.zip" ;;
+  esac
+  curl -o z3.zip -sL "https://github.com/Z3Prover/z3/releases/download/z3-$Z3_VERSION/z3-$Z3_VERSION-x64-$file"
+
+  if $IS_WIN; then 7z x -bd z3.zip; else unzip z3.zip; fi
+  cp z3-*/bin/z3$EXT $BIN/z3$EXT
+  $IS_WIN || chmod +x $BIN/z3
+  rm z3.zip
+}
+
+install_cvc4() {
+  is_exe "$BIN" "cvc4" && return
+  version="${CVC4_VERSION#4.}" # 4.y.z -> y.z
+
+  case "$RUNNER_OS" in
+    Linux) file="x86_64-linux-opt" ;;
+    Windows) file="win64-opt.exe" ;;
+    macOS) file="macos-opt" ;;
+  esac
+  # Temporary workaround
+  if [[ "$RUNNER_OS" == "Linux" ]]; then
+    curl -o cvc4$EXT -sL "https://cvc4.cs.stanford.edu/downloads/builds/x86_64-linux-opt/unstable/cvc4-2020-08-18-x86_64-linux-opt"
+  else
+    curl -o cvc4$EXT -sL "https://github.com/CVC4/CVC4/releases/download/$version/cvc4-$version-$file"
+  fi
+  $IS_WIN || chmod +x cvc4$EXT
+  mv cvc4$EXT "$BIN/cvc4$EXT"
+}
+
+install_yices() {
+  is_exe "$BIN" "yices" && return
+  ext=".tar.gz"
+  case "$RUNNER_OS" in
+    Linux) file="pc-linux-gnu-static-gmp.tar.gz" ;;
+    macOS) file="apple-darwin18.7.0-static-gmp.tar.gz" ;;
+    Windows) file="pc-mingw32-static-gmp.zip" && ext=".zip" ;;
+  esac
+  curl -o "yices$ext" -sL "https://yices.csl.sri.com/releases/$YICES_VERSION/yices-$YICES_VERSION-x86_64-$file"
+
+  if $IS_WIN; then
+    7z x -bd "yices$ext"
+    mv "yices-$YICES_VERSION"/bin/*.exe "$BIN"
+  else
+    tar -xzf "yices$ext"
+    pushd "yices-$YICES_VERSION" || exit
+    sudo ./install-yices
+    popd || exit
+  fi
+  rm -rf "yices$ext" "yices-$YICES_VERSION"
+}
+
+install_yasm() {
+  is_exe "$BIN" "yasm" && return
+  if [[ "$RUNNER_OS" = "Linux" ]]; then
+    sudo apt-get update -q && sudo apt-get install -y yasm
+  else
+    brew install yasm
+  fi
+}
+
+build() {
+  cabal v2-update
+  cabal v2-configure -j2 --minimize-conflict-set
+  if ! retry cabal v2-build "$@" saw jss && [[ "$RUNNER_OS" == "macOS" ]]; then
+    echo "Working around a dylib issue on macos by removing the cache and trying again"
+    cabal v2-clean
+    retry cabal v2-build "$@" saw jss
+  fi
+}
+
+build_abc() {
+  arch=X86_64
+  case "$RUNNER_OS" in
+    Linux) os="Linux" ;;
+    macOS) os="OSX" ;;
+    Windows) return ;;
+  esac
+  pushd deps/abcBridge
+  $IS_WIN || scripts/build-abc.sh $arch $os
```

Comment:
This is true. We don't build this on windows currently because it doesn't compile correctly. The is_win check is a leftover from when I was still trying to build everything on windows and hadn't yet given up and replaced the windows case with return.